datahub-web-react/src/app/identity/user/SelectRole.tsx

```diff
@@ -71,14 +75,15 @@ export default function SelectRole({ user, userRoleUrn, selectRoleOptions, refet
                         {NO_ROLE_TEXT}
                     </>
                 }
-                value={currentRole?.urn}
+                value={currentRole}
                 onChange={(e) => onSelectRole(e as string)}
+                color={currentRole === NO_ROLE_URN ? ANTD_GRAY[6] : undefined}
             >
-                {selectOptions()}
+                {selectOptions}
             </RoleSelect>
             <AssignRoleConfirmation
                 visible={isViewingAssignRole}
-                roleToAssign={currentRole}
+                roleToAssign={rolesMap.get(currentRole)}
```

Comment:
What happens if we do rolesMap.get(NO_ROLE_URN) returns undefined?

Reply:
that's how it was working previously and if it returns undefined then we know they're trying to remove the role (this will only happen for changing to "No Role")

Reply:
Okay got it - Cool!